forge-gui/res/cardsfolder/upcoming/rottenmouth_viper.txt

```diff
@@ -0,0 +1,19 @@
+Name:Rottenmouth Viper
+ManaCost:5 B
+Types:Creature Elemental Snake
+PT:6/6
+A:SP$ PermanentNonland | Cost$ 5 B Sac<X/Permanent.nonLand/nonland permanent> | AILogic$ SacToReduceCost
```

Comment:
@tool4ever can we try to make this `Sac<X/Permanent.nonLand/nonland permanent>` as Additional Cost?

Reply:
hmn I guess it could be scripted more like the March cycle?
though this would still require the extra spell to use Announce$ instead
but in either case the API should be `PermanentCreature`

Reply:
I need to check if we could add an Announce on the RaiseCost/ReduceCost so we don't need that extra Spell


Also, this might be more interesting, such Combined Reduce Cost (add other cost) seems to apply AFTER the normal RaiseCost effects (so you could reduce them too)

Reply:
> Also, this might be more interesting, such Combined Reduce Cost (add other cost) seems to apply AFTER the normal RaiseCost effects (so you could reduce them too)

What you mean? This should already work?

Reply:
If the ReplaceCost  depends on the Maximum of the Raise Mana Cost like #5675 , it would matter in which order you apply them

Reply:
I used Torgaar as a basis.

Reply:
Why is PermanentCreature inappropriate for a creature spell? :p

Reply:
@Glorax can the Type be changed into PermanentCreature?

Reply:
Yes.